hexa/datasets/queue.py

```diff
@@ -1,18 +1,81 @@
 from logging import getLogger
 
+import pandas as pd
+from django.conf import settings
 from dpq.queue import AtLeastOnceQueue
 
-from hexa.datasets.models import DatasetFileMetadataJob
+from hexa.datasets.api import generate_download_url
+from hexa.datasets.models import (
+    DatasetFileMetadata,
+    DatasetFileMetadataJob,
+    DatasetVersionFile,
+)
 
 logger = getLogger(__name__)
 
 
+def read_file_content(download_url: str, content_type: str) -> pd.DataFrame:
+    try:
+        if content_type == "text/csv":
+            return pd.read_csv(download_url)
+        elif content_type == "application/octet-stream":
+            return pd.read_parquet(download_url)
+        else:
+            raise ValueError(f"Unsupported content type: {content_type}")
+    except pd.errors.ParserError as e:
+        print(f"Error parsing the file content: {e}")
+        return pd.DataFrame()
+    except ValueError as e:
+        print(f"Unsupported file content: {e}")
+        return pd.DataFrame()
+
+
 def create_dataset_file_metadata_task(
     queue: AtLeastOnceQueue, job: DatasetFileMetadataJob
 ):
-    # TODO: imlpement ticket PATHWAYS-98  - extract data in background task
-    dataset_version_file_id = job.args["fileId"]
-    logger.info(f"Creating dataset version file {dataset_version_file_id}")
+    try:
+        dataset_version_file_id = job.args["file_id"]
+        dataset_version_file = DatasetVersionFile.objects.get(
+            id=dataset_version_file_id
+        )
+        logger.info(
+            f"Creating dataset snapshot for version file {dataset_version_file_id}"
+        )
+        dataset_file_metadata = DatasetFileMetadata.objects.create(
+            dataset_version_file=dataset_version_file,
+            status=DatasetFileMetadata.STATUS_PROCESSING,
+        )
+
+        download_url = generate_download_url(dataset_version_file)
+        file_snapshot_df = read_file_content(
+            download_url, dataset_version_file.content_type
+        )
+        if not file_snapshot_df.empty:
```

Comment:
What if the file is empty? We could maybe set the sample to [] no ?

Reply:
I though that setting default  to empty list for the model does already cover it